src/ucp/api/ucp.h

```diff
@@ -1272,9 +1286,37 @@ typedef struct ucp_worker_params {
      * @a ucp_am_recv_callback_t.
      */
     size_t                  am_alignment;
+
+    /**
+     * Worker id. The value has to be unique across other workers connected to
```

Comment:
Not sure how to parse `The value has to be unique across other workers connected to  this one` - does it mean that you have to do some sort of global synchronization to set the unique ID ?

Reply:
sometimes you do not need to do additional sync, like in MPI you have rank id + comm

Reply:
@shamisp, is it good to go?

Reply:
Does the value has to be unique or it is up to the user ? What will happen if it is not unique (will it break internals of UCP ? Do we actually validate that it is unique )? If it only used for tools, it is similar to the `name` ? Why we need id in addition to *name. thanks

Reply:
Yes, it has to be unique and we do not validate it. We do not pack name to the address, but we pack uuid. We want to provide an API for matching peers by the worker id

Reply:
So, it is not for profiling tools but for something else  ? Sorry, I'm a bit confused now. Is the idea that user somehow will specify what worker it wants to connect under the assumption that user defined and has full control over the ID and essentially will the ID as addressing workers globally ? We will need a bit of details on the use case. In theory if the names are unique you can auto-generate the unique ID.

Reply:
`If not specified, the uid is generated internally.` - how do you generate unique Id internally ?

Reply:
IMHO, You can use `ucp_address_pack` combined with `name` to autogenerate `int` value that you can use for identification.

Reply:
@shamisp ability to set worker_uid is needed e.g. for spark. On the Listener conn_request we need to distinguish somehow peers. Each process in spark has it's unique id, so if we can set it on a client and query on a server we'll know that this conn_request came from process PID. #6859

Reply:
So, the goal is to append "some" value to the address so it can be exchanged through CM ? The ID is then used for connection matching ?

Reply:
Like socket ?